### PR TITLE
Fix setting breakpoints and better symbol processing

### PR DESCRIPTION
**Pull request by paaco**
Hi Roland, 

I've created a few fixes this weekend for some long standing issues I encountered while using vs64 last months.

Most of them are pretty straightforward, but the change related to "capitalization in DebugInfo" requires some explanation.
I had issues setting breakpoints during development. Sometimes it worked, sometimes not.
In DebugInfo the full path of each file with uppercased driveletter is used as an [index]. Later on the debug adapter also receives a full path but without this uppercase driveletter. This causes another array to be used to lookup the address for an breakpoint line. This fails miserably if there are multiple !source includes because this 2nd array has different ranges of linenumbers.
Uppercasing the driveletter here fixed this.

You also asked me before to remove the replace(':','') and add proper parsing. So I did, but this complicated some issues down the line.
For instance, the parser throws away whitespace. If there is a line that starts with "10 cafe" in the report, the parser cannot determine if this is a label "cafe" or address $cafe at this point. It picked address. The same occured with an actual define I had in my code "AD=5". This was also turned into an address line.
Actually in the report, the address would be at a different position than the label.

Since a label is not required to start with `.` or `@` and does not necessarily end with `:` the current behavior misses a lot of cases.
Actually acme allows just about anything before an instruction. Therefore I now assume anything is a label except instruction names.

If the data size is not known, it now shows both 8 and 16 bits values in the form: 12 / 1234.
A nice side effect is that you now can type in an address watch like $00FB and have it immediately show its pointer address from $FB/$FC.

Overall I believe vs64 really would benefit from these changes so I hope you will integrate them.

Regards,
Alexander

### CODE REVIEW

File: src/debug_info.js [hunk 1]
```diff
@@ -379,48 +397,62 @@ class DebugInfo {
...
+        if (token.charAt(0) == '.' || token.charAt(0) == '@' || token.charAt(0) == '_') {
+            return true;
+        }
+        var firstChar = token.charAt(0).toLowerCase();
+        if (firstChar < 'a' || firstChar > 'z') {
+            return false;
+        }
+        // instructions cannot be symbols or labels
+        if (token.length == 3
+            &&  ("adc,and,asl,bcc,bcs,beq,bit,bmi,bne,bpl,brk,bvc,bvs,clc,cld,cli,clv,cmp,cpx,cpy,dec,dex,"+
```
Comment (rosc77) [inline src/debug_info.js:414]: Minor, but this might be better put to a global variable.

**Review by rosc77 — APPROVED** [review]

**Comment by rosc77** [comment]
Hi Alexander,

Thanks for the great contribution!

Unfortunately, I do not have time at the moment to test the changes. Would be great if you could support me with that - before releasing on the marketplace, but maybe that can wait for a while.

Roland

File: src/debug_info.js (hunk 1 above)
Comment (paaco) [reply src/debug_info.js:414]: OK

**Comment by paaco** [comment]
Thanks for integrating so quickly.

I created a local vsix to test with myself. That appears to be working well on my existing asm projects.
It finds a lot more symbols and I could put breakpoints and hit them.
In the 0.1.12 version I could set breakpoints but never got them to hit because they were internally set at a completely different place.

But while preparing a test asm file for you to show some of the issues I came to the conclusion that my last change [d90a6b4](https://github.com/rosc77/vs64/pull/20/commits/d90a6b4232d66b7d2aa5a231b8dc63f59fcf7d9c) is not really working as intended:

Converting a StatementType.ADDRESS into a StatementType.LABEL has the effect of dropping the symbol from the Symbols list on the left (intended). But it also makes impossible to put a breakpoint on that line (not intended).
So that has to go out or be changed at least. I'll have a look into that later.

**Pull request merged**